spec/cookbook_spec.rb

```diff
@@ -167,21 +207,76 @@
           :path => 'OWNERS',
         },
       ],
+      :result => [],
+      :result_with_symlink_tracking => [],
+    },
+    {
+      :name => 'when metadata file is not in the root of the cb dir',
+      :files => [
+        {
+          :status => :deleted,
+          :path => 'cookbooks/one/cb_one/files/default/metadata.rb',
+        },
+      ],
       :result => [
+        ['cb_one', :modified],
       ],
+      :result_with_symlink_tracking => [
+        ['cb_one', :modified],
+        ['cb_one', :modified],
+      ],
+    },
+    {
+      :name => 'new symlink being created',
+      :files => [
+        {
+          :status => :modified,
+          :path => 'cookbooks/three/cb_one',
+        },
+      ],
+      :result => [],
+      :result_with_symlink_tracking => [['cb_one', :modified]],
     },
   ]
 
-  fixtures.each do |fixture|
-    it "should handle #{fixture[:name]}" do
-      BetweenMeals::Changes::Cookbook.find(
-        fixture[:files],
-        cookbook_dirs,
-        logger,
-      ).map do |cb|
-        [cb.name, cb.status]
-      end.
-        should eq(fixture[:result])
+  {
+    'Normally' => false,
+    'With symlinks' => true,
+  }.each do |c, track_symlinks|
+    context "Running BetweenMeals #{c}" do
```

Comment:
This looks better, but perhaps we should call `uniq` on these things so we don't need to double-specify the results of the majority of these.

Reply:
when track_symlinks is set we mock a symlink from /cookbooks/three/cb_one to /cookbooks/one/cb_one and this re-tests each of the use cases

Reply:
Right but:
1. we shouldn't double-upload CBs, will we?
2. that's not a very interesting test case. What's more interesting is we're ONLY looking at cookbooks/three/ and not cookbooks/one/ and THEN that symlink is there, do we still see that cookbook?

Reply:
There is no point to having a cookbook in two places. In normal circumstances you would only include /cookbooks/one or /cookbooks/three into your cookbook_dirs array.

But when you have /cookbooks/one and /cookbooks/three in git and you had cookbooks/three/cb_one symlink to cookbooks/one/cb_one you should see the same result for anything that happens to cookbooks/one/cb_one should also happen to cookbooks/three/cb_one.

I think what i will do is remove cookbooks/one/  from the cookbook_dir array during the symlink test.